apps/antalmanac/src/stores/AppStore.ts

```diff
@@ -299,6 +341,29 @@ class AppStore extends EventEmitter {
 
     termsInSchedule = (term: string) =>
         new Set([term, ...this.schedule.getCurrentCourses().map((course) => course.term)]);
+
+    autoSaveSchedule = (action: any[] = [null, null, null]) => {
+        const autoSave = typeof Storage !== 'undefined' && window.localStorage.getItem('autoSave') == 'true';
+        if (autoSave) {
+            const savedUserID = window.localStorage.getItem('userID');
+
+            if (savedUserID != null) {
+                saveSchedule(savedUserID, true, true).then((r) => void r);
+                this.unsavedChanges = false;
+            }
+        } else {
+            if (window.localStorage.getItem('unsavedAction') == null) {
+                const unsavedAction = [action];
+                localStorage.setItem('unsavedAction', JSON.stringify(unsavedAction));
+            } else {
+                // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+                // @ts-ignore
+                const unsavedAction = JSON.parse(window.localStorage.getItem('unsavedAction'));
+                unsavedAction.push(action);
+                localStorage.setItem('unsavedAction', JSON.stringify(unsavedAction));
```

Comment:
I'd rather not use `@ts-ignore` here; Would prefer to rewrite this to handle the `null` more gracefully, like you do above on line 350.